metis-enrichment/metis-enrichment-rest/src/main/java/eu/europeana/enrichment/rest/config/Application.java

```diff
@@ -99,24 +84,18 @@ public void addResourceHandlers(ResourceHandlerRegistry registry) {
   }
 
   @Bean
-  @DependsOn("redisInternalEnricher")
-  Enricher enricher() {
-    return new Enricher(getRedisInternalEnricher());
+  EnrichmentService getEnrichmentService(EnrichmentDao enrichmentDao) {
+    return new EnrichmentService(enrichmentDao);
   }
 
   @Bean
-  Converter converter() {
-    return new Converter();
-  }
-
-  @Bean(name = "redisInternalEnricher")
-  RedisInternalEnricher getRedisInternalEnricher() {
+  EnrichmentDao getEnrichmentDao() {
     final MongoProperties<IllegalArgumentException> mongoProperties = new MongoProperties<>(
         IllegalArgumentException::new);
-    mongoProperties.setMongoHosts(new String[]{enrichmentMongoHost}, new int[]{enrichmentMongoPort});
-    final MongoClient mongoClient = new MongoClientProvider<>(mongoProperties).createMongoClient();
-    final EntityDao entityDao = new EntityDao(mongoClient, enrichmentMongoDatabase);
-    return new RedisInternalEnricher(entityDao, redisProvider);
+    mongoProperties
+        .setMongoHosts(new String[]{enrichmentMongoHost}, new int[]{enrichmentMongoPort});
+    mongoClient = new MongoClientProvider<>(mongoProperties).createMongoClient();
```

Comment:
I guess we know that this is only called once? Maybe it is more clear to give the mongo client its own bean or a if not null check?

Reply:
We could do that, I think spring will close it automatically too.